core/arch/arm/plat-marvell/main.c

```diff
@@ -108,3 +114,36 @@ void console_init(void)
 #endif
 	register_serial_console(&console_data.chip);
 }
+
+#ifdef CFG_HW_UNQ_KEY_SUPPORT
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	uint8_t hw_unq_key[sizeof(hwkey->data)] __aligned(64);
```

Comment:
Why `__aligned(64)`?

Reply:
This will insure the compiler to allocate the variable on a 64-byte boundary, this can make copy operations more efficient, so that compiler can use whatever instructions copy the biggest chunks of memory when performing copies to or from the variable.

Reply:
> Why `__aligned(64)`?

This will insure the compiler to allocate the variable on a 64-byte boundary, this can make copy operations more efficient, so that compiler can use whatever instructions copy the biggest chunks of memory when performing copies to or from the variable.

Reply:
I don't think we need this kind of micro optimization. By the way, which instruction needs a 64-byte aligned address?

Reply:
My understanding is, memcpy performance depends on many factors and various methods to optimize it. Address alignment is the one the base factor. For example, load and store addresses should be aligned to cache lines to allow more efficient memory accesses.
Reference links: 
https://developer.arm.com/documentation/102159/0400/Load-and-store---leftovers
https://www.nxp.com/docs/en/application-note/AN12628.pdf   (2.2.1 Address align)

Reply:
This is undoubtedly true, but why optimize for speed here? And even so, would `__aligned(64)` make a measurable difference? Have you run benchmarks? :wink:
We are certainly not going to add alignment directives to all variables passed to `memcpy()`.

Reply:
> This is undoubtedly true, but why optimize for speed here? And even so, would `__aligned(64)` make a measurable difference? Have you run benchmarks? 😉
> We are certainly not going to add alignment directives to all variables passed to `memcpy()`.

Not really measured the performance difference. Okay, I will remove the directive.